File: backend/NoPressure.Common/DTO/ActivityDTO.cs
using NoPressure.Common.Enums;

namespace NoPressure.Common.DTO
{
    public class ActivityDTO
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public string Color { get; set; }

        public ScheduleHour StartTime { get; set; }
        public ScheduleHour EndTime { get; set; }
        public DateTime Date { get; set; }

        public bool IsRepeatable { get; set; }
        public ActivityState State { get; set; }
        public ScheduleHour DirectiveTerm { get; set; }
        public int Priority { get; set; }
        public double DelayCoefficient { get; set; }
        public int Duration { get; set; }
    }
}

File: frontend/src/app/models/activity/activity-dto.ts
import { ActivityState } from "../enums/ActivityState";
import { ScheduleHour } from "../enums/ScheduleHour";

export interface ActivityDTO {
    id: number;
    userId: number;
    name: string;
    description: string;
    color: string;
    state: ActivityState;
    startTime?: ScheduleHour;
    endTime?: ScheduleHour;
}